gradle/mockito-core/testing.gradle

```diff
@@ -6,24 +6,43 @@ if (java11 != null) {
     }
 }
 
-task(createTestResources).doLast {
-    def mockMakerFile = new File("$projectDir/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker")
-    if (System.env.MOCK_MAKER != null) {
-        logger.info("Using MockMaker ${System.env.MOCK_MAKER}")
-        mockMakerFile.parentFile.mkdirs()
-        mockMakerFile.createNewFile()
-        mockMakerFile.write(System.env.MOCK_MAKER)
-    } else {
-        logger.info("Using default MockMaker")
+task(createTestResources) {
+    doLast {
+        // Configure MockMaker from environment (if specified), otherwise use default
+        def mockMakerFile = new File("$sourceSets.test.output.resourcesDir/mockito-extensions/org.mockito.plugins.MockMaker")
+        if (System.env.MOCK_MAKER != null && !System.env.MOCK_MAKER.endsWith("default")) {
+            logger.info("Using MockMaker ${System.env.MOCK_MAKER}")
+            mockMakerFile.parentFile.mkdirs()
+            mockMakerFile.createNewFile()
+            mockMakerFile.write(System.env.MOCK_MAKER)
+        } else {
+            logger.info("Using default MockMaker")
+        }
+
+        // Configure MemberAccessor from environment (if specified), otherwise use default
+        def memberAccessorFile = new File("$sourceSets.test.output.resourcesDir/mockito-extensions/org.mockito.plugins.MemberAccessor")
+        if (System.env.MEMBER_ACCESSOR != null  && !System.env.MEMBER_ACCESSOR.endsWith("default")) {
+            logger.info("Using MemberAccessor ${System.env.MEMBER_ACCESSOR}")
+            memberAccessorFile.parentFile.mkdirs()
+            memberAccessorFile.createNewFile()
+            memberAccessorFile.write(System.env.MEMBER_ACCESSOR)
+        } else {
+            logger.info("Using default MemberAccessor")
+        }
     }
 }
 
 task(removeTestResources).doLast {
-    def mockMakerFile = new File("$projectDir/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker")
+    def mockMakerFile = new File("$sourceSets.test.output.resourcesDir/mockito-extensions/org.mockito.plugins.MockMaker")
     if (mockMakerFile.exists()) {
         mockMakerFile.delete()
     }
+
+    def memberAccessorFile = new File("$sourceSets.test.output.resourcesDir/mockito-extensions/org.mockito.plugins.MemberAccessor")
+    if (memberAccessorFile.exists()) {
+        memberAccessorFile.delete()
+    }
 }
 
-compileTestJava.dependsOn createTestResources
-compileTestJava.finalizedBy removeTestResources
+processTestResources.finalizedBy(createTestResources)
```

Comment:
@TimvdLippe the problem turned out to be worth than initially suggested: not only `MemberAccessor` were not picked, none of the `MockMaker`s were exercised except the default one.

What was happening is that the `mockito-extensions` created by `createTestResources` were immediately deleted by `removeTestResources` upon `compileTestJava` finalization. Essentially, all tests were run with default `MockMaker` only.

I fixed that but large number of tests are not compatible with `ProxyMockMaker` because they mock classes, not interfaces. I have added the large number of exclusions like:

```
    @BeforeClass
    public static void check_mock_maker() {
        Assume.assumeThat(Plugins.getMockMaker(), not(instanceOf(ProxyMockMaker.class)));
    }
```

Do you think this is fine? I have ~40 test cases left to troubleshoot

Reply:
Hm, that's a good point. If so many tests from core fail with the proxy mockmaker, I don't think that's worth it. In that case, let's remove it from our build matrix. The project itself should have sufficient coverage then.